File: 0179-largest-number/0179-largest-number.cpp
class Solution {
public:
    string largestNumber(vector<int>& nums) {
        // Convert all integers to strings
        vector<string> strNums;
        for (int num : nums) {
            strNums.push_back(to_string(num));
        }

        // Define a custom comparator lambda function
        auto compare = [](const string& a, const string& b) {
            return a + b > b + a; // If concatenating a+b is greater than b+a, a should come before b
        };

        // Sort the numbers using the custom comparator
        sort(strNums.begin(), strNums.end(), compare);

        // Concatenate the sorted strings
        string result;
        for (const string& s : strNums) {
            result += s;
        }

        // Handle the edge case where the result might be all zeros
        if (result[0] == '0') {
            return "0";
        }

        return result;
    }
};

File: 0179-largest-number/0179-largest-number.py
class Solution:
    def largestNumber(self, nums: List[int]) -> str:
        nums = [str(n) for n in nums]

        for i in range(len(nums)):
            for j in range(1, len(nums)):
                if nums[j-1] + nums[j] > nums[j] + nums[j-1]:
                    nums[j-1], nums[j] = nums[j], nums[j-1]

        nums.reverse()
        return str(int("".join(nums)))